### PR TITLE
[590] Add RFC template for XTable

### PR DESCRIPTION
**Pull request by vinishjail97**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

Introduce RFC template to propose/review new technical specs in the project.  

## Brief change log

*(for example:)*
- *Add RFC template*

## Verify this pull request

*(Please pick either of the following options)*

This pull request is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: rfc/template.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Abstract
+
+Describe the problem you are trying to solve and a brief description of why it’s needed
+
+## Background
+Introduce any much background context which is relevant or necessary to understand the feature and design choices.
+
+## Implementation
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture.
+Provide a detailed description of how you intend to implement this feature.This may be fairly extensive and have large subsections of its own.
```
Comment (the-other-tim-brown) [inline rfc/template.md:43]: You need to add `\` at the end of a line or else it is rendering as `own.Or`

File: rfc/template.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Or it may be a few sentences. Use judgement based on the scope of the change.
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
```
Comment (the-other-tim-brown) [inline rfc/template.md:48]: Should we call out any breakage to existing APIs?

File: rfc/template.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Status
+
+GH Feature Request: <link to umbrella JIRA>
+
+> Please keep the status updated in `rfc/README.md`.
+
+## Abstract
+
+Describe the problem you are trying to solve and a brief description of why it’s needed
```
Comment (the-other-tim-brown) [inline rfc/template.md:36]: Nitpick: add a `.`

File: rfc/template.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Background
+Introduce any much background context which is relevant or necessary to understand the feature and design choices.
```
Comment (the-other-tim-brown) [inline rfc/template.md:39]: `any much` -> `any`

File: rfc/template.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture.
+Provide a detailed description of how you intend to implement this feature.This may be fairly extensive and have large subsections of its own.
+Or it may be a few sentences. Use judgement based on the scope of the change.
+
+## Rollout/Adoption Plan
+
+- What impact (if any) will there be on existing users?
+- If we are changing behavior how will we phase out the older behavior?
+- If we need special migration tools, describe them here.
+- When will we remove the existing behavior
```
Comment (the-other-tim-brown) [inline rfc/template.md:51]: Combine this with 49?

File: rfc/template.md (hunk 2 above)
Comment (vinishjail97) [reply rfc/template.md:48]: yeah makes sense, this question seems to be overloaded.

File: rfc/template.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+Introduce any background context which is relevant or necessary to understand the feature and design choices.
+
+## Implementation
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture.
+Provide a detailed description of how you intend to implement this feature.This may be fairly extensive and have large subsections of its own or it may be a few sentences. 
+Use judgement based on the scope of the change.
+
+## Rollout/Adoption Plan
+
+- Are there any breaking changes as part of this new feature/functionality ?
```
Comment (the-other-tim-brown) [inline rfc/template.md:48]: are the spaces before the `?` required in markdown?

File: rfc/template.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+## Abstract
+
+Describe the problem you are trying to solve and a brief description of why it’s needed.
+
+## Background
+Introduce any background context which is relevant or necessary to understand the feature and design choices.
+
+## Implementation
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture.
+Provide a detailed description of how you intend to implement this feature.This may be fairly extensive and have large subsections of its own or it may be a few sentences.
```
Comment (the-other-tim-brown) [inline rfc/template.md:43]: Put a space before the periods

**Review by the-other-tim-brown — COMMENTED** [review]
@vinishjail97 please take a look at this in preview mode and clean up the formatting issues

File: rfc/template.md (hunk 7 above)
Comment (vinishjail97) [reply rfc/template.md]: Used line breaks to be more clear.

File: rfc/template.md (hunk 6 above)
Comment (vinishjail97) [reply rfc/template.md]: No it's not required.

**Comment by vinishjail97** [comment]
> @vinishjail97 please take a look at this in preview mode and clean up the formatting issues

I have fixed the formatting, please take a look again.

File: rfc/template.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+Describe the problem you are trying to solve and a brief description of why it’s needed.
+
+## Background
+Introduce any background context which is relevant or necessary to understand the feature and design choices.
+
+## Implementation
+Describe the new thing you want to do in appropriate detail, how it fits into the project architecture.<br>
+Provide a detailed description of how you intend to implement this feature, this may be fairly extensive and have large subsections of its own or it may be a few sentences.<br> 
+Use judgement based on the scope of the change.
```
Comment (the-other-tim-brown) [inline rfc/template.md:44]: Use judgement for what? Is this meant for the line above?

Comment (vinishjail97) [reply rfc/template.md:44]: Use judgement on how detailed the description needs to be.

Comment (vinishjail97) [reply rfc/template.md]: I have modified the statement and added this - `If unclear, you can ask questions in dev@xtable.apache.org/slack.`

**Review by the-other-tim-brown — APPROVED** [review]

**Pull request merged**